data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -46,14 +47,17 @@ public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         String url = csb.getClusterUrl();
         URI clusterUri = new URI(url);
         String host = clusterUri.getHost();
+        Ensure.argIsNotNull(clusterUri.getAuthority(), "clusterUri.authority");
```

Comment:
This is nice in C#, but Java already gives us `Objects.requireNonNull()` and Apache Commons already gives us `Validate.notNull()`

I'd get rid of this Ensure class.

Reply:
I see you didn't create it, and it's pretty entrenched in our project. I guess we can't get rid of it altogether, but I personally would reduce our dependency on it and use the built-in Java method here.